### PR TITLE
packaging updates

### PR DESCRIPTION
**Pull request by arnetheduck**
* one package per binary (nimbus_beacon_node, nimbus_validator_client)
* use `-` in package name (`_` is separating the version)
* don't include (un)installation scripts in package
* default metrics port 8009 for vc
* fix several upgrade/install errors in scripts
* add JWT option to service files

### CODE REVIEW

File: scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service [hunk 1]
```diff
@@ -27,8 +29,10 @@ Environment=NETWORK=mainnet
 
 # You need to have have access to an execution client - by default, we assume
 # a compatible execution client is running on the same machine on the default
-# websocket port.
+# websocket port. Make sure to supply the right path to the JWT secret.
+
 Environment=WEB3_URL=ws://127.0.0.1:8546
+Environment=JWT_SECRET=/tmp/jwtsecret
```
Comment (zah) [inline scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service:35]: The `/tmp` folder is a system-wide namespace and `jwtsecret` is too generic as a name. We should use a prefix like `nimbus` in the filename.

File: scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service [hunk 2]
```diff
@@ -51,7 +55,8 @@ TimeoutSec=1200
 Restart=always
 
 # Don't restart when Doppelganger detection has been activated
-RestartPreventExitStatus=1031
+# TODO https://github.com/status-im/nimbus-eth2/issues/3973
+# RestartPreventExitStatus=1031
```
Comment (zah) [inline scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service]: I've already submitted a PR changing this here:
https://github.com/status-im/nimbus-eth2/pull/3977

File: scripts/package_src/nimbus_validator_client/after_install [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -e
+
+DISTRO=$(lsb_release -si)
+if ! id -u nimbus > /dev/null 2>&1; then
+  case $DISTRO in
+    Ubuntu|Debian)
+      adduser --system --no-create-home --group nimbus
```
Comment (zah) [inline scripts/package_src/nimbus_validator_client/after_install:10]: `adduser` will fail if the user `nimbus` already exists, which will result in a failing installation due to the `set -e` directive above.

Here are some ideas how to make the command idempotent:
https://unix.stackexchange.com/questions/28526/add-a-user-to-the-system-only-if-it-doesnt-exist

File: scripts/package_src/nimbus_validator_client/deb_after_purge [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+set -e
+
+# TODO We purge the nimbus user / group / folder from the beacon node uninstall
```
Comment (zah) [inline scripts/package_src/nimbus_validator_client/deb_after_purge]: Right, this looks like an open question.

File: scripts/package_src/nimbus_validator_client/image/lib/systemd/system/nimbus_validator_client.service [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+Environment=METRICS_PORT=8009
+
+# Interaction and monitoring
+Environment=METRICS_ENABLED=Yes
+
+# Default group = nimbus
+User=nimbus
+WorkingDirectory=/var/lib/nimbus
+TimeoutSec=1200
+Restart=always
```
Comment (zah) [inline scripts/package_src/nimbus_validator_client/image/lib/systemd/system/nimbus_validator_client.service:46]: The validator client support the doppelganger exit code as well.

File: beacon_chain/conf.nim [hunk 6]
```diff
@@ -841,7 +841,7 @@ type
 
     metricsPort* {.
       desc: "Listening HTTP port of the metrics server"
-      defaultValue: 8008
+      defaultValue: 8009
```
Comment (zah) [inline beacon_chain/conf.nim]: How about a port like `8108` to make it easier to use our current practice of creating scripts that use the `BASE_PORT + NODE_ID` pattern?

File: scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service (hunk 1 above)
Comment (arnetheduck) [reply scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service:35]: the jwtsecret is owned by the execution client (it's the EC that creates it), and both nethermind, geth and several other guides use `/tmp/jwtsecret` in their [documentation](https://geth.ethereum.org/docs/interface/consensus-clients) - see the example command line.

File: scripts/package_src/nimbus_validator_client/after_install (hunk 3 above)
Comment (arnetheduck) [reply scripts/package_src/nimbus_validator_client/after_install:10]: hm, so according to the debian manpage, `adduser` returns 0 if the user already exists: https://manpages.debian.org/jessie/adduser/adduser.8.en.html

this is contrary to `useradd`, which indeed gives an error :/ it also ties in with the TODO as to which package should be responsible for the user - if all packages create-if-don't-exist `nimbus`, none should remove it until all are removed - it's a messy problem.

finally, `useradd` actually exists on debian - `adduser` is just a wrapper - we could simply use `useradd` always, and then with the `id` trick..

Comment (arnetheduck) [reply scripts/package_src/nimbus_validator_client/after_install:10]: btw, the whole section is already gated by an `if ! id` check above, meaning it already checks if the user exists

File: scripts/package_src/nimbus_validator_client/deb_after_purge (hunk 4 above)
Comment (arnetheduck) [reply scripts/package_src/nimbus_validator_client/deb_after_purge]: actually, turns out it it isn't - most guides recommend not to remove created users

File: scripts/package_src/nimbus_validator_client/image/lib/systemd/system/nimbus_validator_client.service (hunk 5 above)
Comment (arnetheduck) [reply scripts/package_src/nimbus_validator_client/image/lib/systemd/system/nimbus_validator_client.service:46]: rebased

File: beacon_chain/conf.nim (hunk 6 above)
Comment (arnetheduck) [reply beacon_chain/conf.nim]: done

File: scripts/package_src/nimbus_validator_client/after_install (hunk 3 above)
Comment (zah) [reply scripts/package_src/nimbus_validator_client/after_install:10]: Ah, I've missed that, sorry.

File: scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service (hunk 1 above)
Comment (zah) [reply scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service:35]: That's still quite ridiculous, but, oh, well...

File: scripts/package_src/nimbus_beacon_node/before_remove [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+set -e
+
+SERVICE_NAME="nimbus_beacon_node"
+
+if systemctl --all --type service | grep -q "$SERVICE_NAME";then
```
Comment (zah) [inline scripts/package_src/nimbus_beacon_node/before_remove]:
```suggestion
if systemctl --all --type service | grep -q "$SERVICE_NAME"; then
```

**Pull request merged**